src/reducer/admin/resource/index.js

```diff
@@ -1,9 +1,37 @@
-import { combineReducers } from 'redux';
+import { DECLARE_RESOURCES } from '../../../actions';
+
 import data from './data';
 import list from './list';
 
-export default resource =>
-    combineReducers({
-        data: data(resource),
-        list: list(resource),
-    });
+const initialState = {};
+export default (previousState = initialState, action) => {
+    if (action.type === DECLARE_RESOURCES) {
+        const newState = action.payload.reduce(
+            (acc, resource) => ({
+                ...acc,
+                [resource.name]: {
+                    props: resource,
+                    data: data(resource.name)(undefined, action),
+                    list: list(resource.name)(undefined, action),
+                },
+            }),
+            {}
+        );
+        return newState;
+    }
+
+    const resources = Object.keys(previousState);
+    const newState = resources.reduce(
```

Comment:
Yes but as we only pass them to the inner reducers, that should not add much overhead. If you think it's necessary, I could add some check on the meta as the inner reducers do